### ISSUE TITLE
Formatting RDF Lists results in invalid output

### ISSUE THREAD

**Issue by fkleedorfer** [atextor/turtle-formatter#13]
Using default style, 

```
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix qudt: <http://qudt.org/schema/qudt/> .
@prefix sh: <http://www.w3.org/ns/shacl#> .

qudt:IntegerUnionList
  a rdf:List ;
  rdf:first [
      sh:datatype xsd:nonNegativeInteger ;
    ] ;
  rdf:rest (
      [
        sh:datatype xsd:positiveInteger ;
      ]
      [
        sh:datatype xsd:integer ;
      ]
    ) ;
  rdfs:label "Integer union list" ;
.
```
is transformed to 
```
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix qudt: <http://qudt.org/schema/qudt/> .
@prefix sh: <http://www.w3.org/ns/shacl#> .

 ( [
  sh:datatype xsd:nonNegativeInteger ;
] [
  sh:datatype xsd:positiveInteger ;
] [
  sh:datatype xsd:integer ;
] ) a rdf:List ;
  rdfs:label "Integer union list" ;
  rdf:first;
  rdf:rest ( ) .
```

which is not only wrong but also not valid TTL anymore.

**Comment by fkleedorfer** [comment]
Seems like the formatting code wants to treat the list as an anonymous list, which it is not...

**Comment by atextor** [comment]
Hi @fkleedorfer,
you are correct, the code assumes an RDF list is always anonymous and does not use predicates other than `rdf:rest` and `rdf:first`. I personally consider adding other predicates to list nodes a code smell (because then you can't use the `( )` syntax in Turtle any more) but of course it must still be formatted in a valid way.

**Comment by fkleedorfer** [comment]
Ok. Fixed in #15.
